frontend/rust-lib/flowy-database/src/services/export.rs

```diff
@@ -0,0 +1,93 @@
+use crate::entities::FieldType;
+use crate::services::cell::{stringify_cell_data, TypeCellData};
+use crate::services::database::DatabaseEditor;
+use database_model::FieldRevision;
+use flowy_error::{FlowyError, FlowyResult};
+use serde::Serialize;
+use std::collections::HashMap;
+use std::sync::Arc;
+
+#[derive(Debug, Clone, Serialize)]
+pub struct ExportField {
+  pub id: String,
+  pub name: String,
+  pub field_type: i64,
+  pub visibility: bool,
+  pub width: i64,
+  pub type_options: HashMap<String, String>,
+  pub is_primary: bool,
+}
+
+impl From<&Arc<FieldRevision>> for ExportField {
+  fn from(field_rev: &Arc<FieldRevision>) -> Self {
+    let type_options = field_rev
+      .type_options
+      .iter()
+      .map(|(k, v)| (k.clone(), v.clone()))
+      .collect();
```

Comment:
should we make the type in type_options copyable and use `.clone` directly instead of map clone?

```rs
type_options.clone()
```

Reply:
Well. You found this. But nope. The type of type_options is IndexMap not HashMap.